opendevin/controller/agent_controller.py

```diff
@@ -170,21 +167,10 @@ async def step(self, i: int):
         try:
             action = self.agent.step(self.state)
             if action is None:
-                raise AgentNoActionError()
-            logger.info(action, extra={'msg_type': 'ACTION'})
-        except Exception as e:
+                raise AgentNoActionError('No action was returned')
+        except (AgentMalformedActionError, AgentNoActionError) as e:
             observation = AgentErrorObservation(str(e))
-            logger.error(e)
-            logger.debug(traceback.format_exc())
-
-            # raise specific exceptions that need to be handled outside
-            # note: we are using classes from openai rather than litellm because:
-            # 1) litellm.exceptions.AuthenticationError is a subclass of openai.AuthenticationError
-            # 2) embeddings call, initiated by llama-index, has no wrapper for errors.
-            #    This means we have to catch individual authentication errors
-            #    from different providers, and OpenAI is one of these.
-            if isinstance(e, (AuthenticationError, ContextWindowExceededError, APIConnectionError)):
-                raise
```

Comment:
I didn't test yet, afaik though, the exception `raised` here is the one caught in the caller to stop the loop, when there's no good api key or other unrecoverable errors. Is the intention to no longer stop the loop for unrecoverable errors?

Reply:
The big change is that now errors get `raised` _by default_! We only catch a specific set of errors (MalformedAction, NoAction), and the rest are treated as unrecoverable and stop the loop.